pkg/plugins/gittag/target.go

```diff
@@ -0,0 +1,100 @@
+	changed, _, _, err = gt.target(source, dryRun)
+
+	return changed, err
+}
+
+// TargetFromSCM create and push a git tag based on the SCM configuration
+func (gt *GitTag) TargetFromSCM(source string, scm scm.ScmHandler, dryRun bool) (changed bool, files []string, message string, err error) {
+	if len(gt.spec.Path) > 0 {
+		logrus.Warningf("Path setting value %q overridden by the scm configuration (value %q)",
+			gt.spec.Path,
+			scm.GetDirectory())
+	}
+	gt.spec.Path = scm.GetDirectory()
+
+	changed, files, message, err = gt.target(source, dryRun)
+	if err != nil {
+		return changed, files, message, err
+	}
+
+	err = scm.PushTag(source)
+	if err != nil {
+		logrus.Errorf("Git push tag error: %s", err)
+		return changed, files, message, err
+	}
+	logrus.Infof("%s The Git Tag %q was pushed successfully to the specified remote.", result.ATTENTION, source)
+	return changed, files, message, err
+}
+
+func (gt *GitTag) target(source string, dryRun bool) (bool, []string, string, error) {
+	// Ensure that a git message is present to annotate the tag to create
+	if len(gt.spec.Message) == 0 {
+		// absence of a message is not blocking: warn the user and continue
+		gt.spec.Message = "Generated by updatecli"
+		logrus.Warningf("No specified message for gittag target. Using default value %q", gt.spec.Message)
+	}
+
+	files := []string{}
+	message := gt.spec.Message
+
+	// Fail if a pattern is specified
+	if gt.spec.VersionFilter.Pattern != "" {
+		return false, files, message, fmt.Errorf("Target validation error: spec.versionfilter.pattern is not allowed for targets of type gittag.")
+	}
+
+	// Fail if the git tag resource cannot be validated
+	err := gt.Validate()
+	if err != nil {
+		logrus.Errorln(err)
+		return false, files, message, err
+	}
+
+	// Check if the provided tag (from source input value) already exist
+	gt.spec.VersionFilter.Pattern = source
+	tags, err := generic.Tags(gt.spec.Path)
+	if err != nil {
+		return false, files, message, err
+	}
+	gt.foundVersion, err = gt.spec.VersionFilter.Search(tags)
+	if err != nil && err.Error() != fmt.Sprintf("No version found matching pattern %q", source) {
+		// Something went wrong during the tag search.
+		return false, files, message, err
+	}
+	if gt.foundVersion.ParsedVersion == source {
+		// No error, but no change
+		logrus.Printf("%s The Git Tag %q already exists, nothing else to do.",
+			result.SUCCESS,
+			source)
+		return false, files, message, nil
+	}
+
+	// Otherwise proceed to creates this new tag
+	logrus.Printf("%s The Git Tag %q does not exist: creating it.", result.ATTENTION, source)
+
+	if dryRun {
+		// Dry run: no changes to apply.
+		// Return early without creating tag but notifies that a change should be made.
+		return true, files, message, nil
+	}
+
+	changed, err := generic.NewTag(source, gt.spec.Message, gt.spec.Path)
+	if err != nil {
+		return changed, files, message, err
+	}
+	logrus.Printf("%s The Git Tag %q was created.", result.ATTENTION, source)
```

Comment:
```suggestion
	logrus.Printf("%s The git tag %q has been created.", result.ATTENTION, source)
```

Reply:
ack, even though I'm not sure if using the preterit form isn't best.